sklearn/decomposition/pca.py

```diff
@@ -362,7 +362,7 @@ def _fit(self, X):
             raise TypeError('PCA does not support sparse input. See '
                             'TruncatedSVD for a possible alternative.')
 
-        X = check_array(X, dtype=[np.float64], ensure_2d=True,
+        X = check_array(X, dtype=[np.float32, np.float64], ensure_2d=True,
```

Comment:
We need to be consistent between estimator:

The other PR use `[np.float64, np.float32]` and not `[np.float32, np.float64]`. It is important since when the dtype is not on the list, it is casted to the first one on the list.